models/application.py

```diff
@@ -41,21 +41,23 @@ def get_by_id(self, app_id):
             'filter': 'id',
             'eq': app_id
         }
-        app = self.base_request.request('application', 'GET', params=params, endpoint=self.settings.get('pine_endpoint'))['d'][0]
-        if app:
-            return app
-        else:
-            # found no application
-            print(Message.NO_APPLICATION_FOUND.format(value=app_id, app_att="id"))
+        try:
+            return self.base_request.request('application', 'GET', params=params, endpoint=self.settings.get('pine_endpoint'))['d'][0]
+        except IndexError:
+            raise ResinException.ResinApplicationNotFound(app_id)
 
     def create(self, name, device_type):
-        device_slug = self.device.get_device_slug(device_type)
+        device_type_found = self.config.get_device_type()
```

Comment:
Why get_device_type returns a list? It would save you the following list comprehension if it didn't.